### PR TITLE
Fix for selecting and deleting subfolders and files

### PR DESCRIPTION
**Pull request by hannyle**
### Description

- Subfolders and files should be all selectable by checking checkbox
- Subfolders shouldn't be deleted before deleting all files inside it
- Files should be able to select and mass delete

### Related issues
GL 933 & 936 & 974

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Changes Made

- Added folder icon for sub-folders
- Added `prefix` to file's names so the `file` names are matched when its checkbox is selected
- Changed `Display as text` to `Display as path` to make the term more understandable
- Prevented `subfolders` from being deleted when clicking `Delete` button or selecting together with `files`
- Updated Changelog

### Testing
- [x] Needs testing (manual test)

### Mention

- Since we have now `subfolders` and `files` in the same view, some notifications when selecting and deleting them are not clear enough and still not in correct timing/visual order
- The terms and text are still not so clear since we only have `objects` before, now `objects` could mean `subfolders` and `files``
--> These things should be fixed in another PR while waiting for the UI design from @ainoc

### CODE REVIEW

**Comment by hannyle** [comment]
Noticed an issue with this, I will make changes

**Comment by hannyle** [comment]
Ready for review

**Comment by hannyle** [comment]
A follow-up PR will be needed, as I wrote in the **Mention** part

File: .github/config/.finnishwords.txt
Comment (emm1R) [inline .github/config/.finnishwords.txt:1]: This is not a word?

Comment (hannyle) [reply .github/config/.finnishwords.txt:1]: You mean `poluna` ? I just google translated  it :D

Comment (emm1R) [reply .github/config/.finnishwords.txt:1]: Then it should be `polkuna`.

**Review by sampsapenna — APPROVED** [review]
Works nicely, and displaying only the paths of the current folder recursively is indeed more intuitive than the old behavior. Uploads into nested folders still works as well, so I didn't notice any regressions.

LGTM 👍🏼

**Pull request merged**